### ISSUE TITLE
Possibility to deactivate vSphere CSI

### ISSUE THREAD

**Issue by steled** [kubermatic/kubeone#2750]
### Description of the feature you would like to add / User story

As a KubeOne user
I would like to have the ability to deactivate the vSphere CSI
in order to be able to deploy a cluster without vSphere CSI

### Solution details

- disable vSphere CSI
- deploy cluster without putting a `csiConfig` to the `KubeOneCluster` configuration

### Use cases

I want to use the vSphere provider but without the vSphere CSI because I want to use Ceph as a storage backend.
The problem is if I only put the following lines of code into the `KubeOneCluster` config is that the pods of the `vsphere-csi-controller` deployment run into a `CrashLoopBackOff`.

```yaml
cloudProvider:
  vsphere: {}
  external: true
  csiConfig: |
    [VirtualCenter "<VCENTER>"]
cloudConfig: |
<omitted for brevity>
```

### Additional information

You can also check https://github.com/kubermatic/kubeone/issues/2737 for some background information.

**Comment by steled** [comment]
I want to use ceph csi but don't know if it would be possible to combine vSphere CCM with a different CSI (for example Ceph).
My understanding is that I need to install a different CSI after cluster deployment manually?!

**Comment by kron4eg** [comment]
> My understanding is that I need to install a different CSI after cluster deployment manually?!

Yep, and assign an StorageClass of ceph-csi as a default one, to make it easier.

**Comment by steled** [comment]
Ok, than it should be fine to just use the vSphere CCM without any CSI.

**Comment by embik** [comment]
Note for someone picking this up: It probably makes sense to make this a generic option for disabling any CSI driver we provide.

**Comment by pascalrimann** [comment]
what is currently the best way to disable the vsphere csi? I have to use Astra Trident CSI

**Comment by embik** [comment]
> what is currently the best way to disable the vsphere csi? I have to use Astra Trident CSI

It's currently not possible, but is being implemented in #2784. If you need to use a different CSI driver, just install it in addition and only use StorageClasses that use that CSI driver.
